Build error on 32-bit architecture due to missing static_casts

### Description

Build error on i386 (ia32):

1.
```
/wrkdirs/usr/ports/audio/mumble-server/work/mumble-1.5.613/src/MumbleProtocol.cpp:784:57: error: implicit conversion loses integer precision: 'std::uint64_t' (aka 'unsigned long long') to 'size_type' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
  784 |                 m_audioData.payload = gsl::span< byte >(payloadBegin, payloadSize);
      |                                       ~~~                             ^~~~~~~~~~~
```
2.
```
/wrkdirs/usr/ports/audio/mumble-server/work/mumble-1.5.613/src/murmur/Server.cpp:1044:21: error: implicit conversion changes signedness: 'int' to 'std::vector<char>::size_type' (aka 'unsigned int') [-Werror,-Wsign-conversion]
                bufVec.resize(len + 4);
                       ~~~~~~ ~~~~^~~
1 error generated.
```
3.
```
/wrkdirs/usr/ports/audio/mumble/work/mumble-1.5.613/src/mumble/Audio.cpp:46:25: error: implicit conversion loses integer precision: 'qint64' (aka 'long long') to 'long' [-Werror,-Wshorten-64-to-32]
   46 |                 long time = qetTicker.elapsed();
      |                      ~~~~   ~~~~~~~~~~^~~~~~~~~
1 error generated.
```
4.
```
/wrkdirs/usr/ports/audio/mumble/work/mumble-1.5.613/src/mumble/AudioOutput.cpp:620:16: error: implicit conversion changes signedness: 'unsigned int' to 'std::ptrdiff_t' (aka 'int') [-Werror,-Wsign-conversion]
  620 |                                                         recbuff[i] += (pfBuffer[2 * i] / 2.0f + pfBuffer[2 * i + 1] / 2.0f) * volumeAdjustment;
      |                                                         ~~~~~~~ ^
/wrkdirs/usr/ports/audio/mumble/work/mumble-1.5.613/src/mumble/AudioOutput.cpp:624:16: error: implicit conversion changes signedness: 'unsigned int' to 'std::ptrdiff_t' (aka 'int') [-Werror,-Wsign-conversion]
  624 |                                                         recbuff[i] += pfBuffer[i] * volumeAdjustment;
      |                                                         ~~~~~~~ ^
2 errors generated.
```
FreeBSD 13.2, 13.3, 14.0, 15-CURRENT (`main` branch).

### Steps to reproduce

1. Build mumble

### Mumble version

1.5.613

### Mumble component

Both

### OS

FreeBSD

### Reproducible?

Yes

### Additional information

_No response_

### Relevant log output

_No response_

### Screenshots

_No response_

--- a/src/MumbleProtocol.cpp
+++ b/src/MumbleProtocol.cpp
@@ -781,7 +781,7 @@ namespace Protocol {
 		}
 
 
-		m_audioData.payload = gsl::span< byte >(payloadBegin, payloadSize);
+		m_audioData.payload = gsl::span< byte >(payloadBegin, static_cast< std::size_t >(payloadSize));
 
 		if (stream.left() == 3 * sizeof(float)) {
 			// If there are further bytes after the audio payload, this means that there is positional data attached to

--- a/src/mumble/Audio.cpp
+++ b/src/mumble/Audio.cpp
@@ -43,7 +43,7 @@ void LoopUser::addFrame(const Mumble::Protocol::AudioData &audioData) {
 		QMutexLocker l(&qmLock);
 		bool restart = (qetLastFetch.elapsed() > 100);
 
-		long time = qetTicker.elapsed();
+		qint64 time = qetTicker.elapsed();
 
 		float r;
 		if (restart)

--- a/src/mumble/AudioOutput.cpp
+++ b/src/mumble/AudioOutput.cpp
@@ -616,12 +616,13 @@ bool AudioOutput::mix(void *outbuff, unsigned int frameCount) {
 					if (speech->bStereo) {
 						// Mix down stereo to mono. TODO: stereo record support
 						// frame: for a stereo stream, the [LR] pair inside ...[LR]LRLRLR.... is a frame
-						for (unsigned int i = 0; i < frameCount; ++i) {
-							recbuff[i] += (pfBuffer[2 * i] / 2.0f + pfBuffer[2 * i + 1] / 2.0f) * volumeAdjustment;
+						for (std::size_t i = 0; i < frameCount; ++i) {
+							recbuff.get()[i] +=
+								(pfBuffer[2 * i] / 2.0f + pfBuffer[2 * i + 1] / 2.0f) * volumeAdjustment;
 						}
 					} else {
-						for (unsigned int i = 0; i < frameCount; ++i) {
-							recbuff[i] += pfBuffer[i] * volumeAdjustment;
+						for (std::size_t i = 0; i < frameCount; ++i) {
+							recbuff.get()[i] += pfBuffer[i] * volumeAdjustment;
 						}
 					}
 

--- a/src/murmur/Server.cpp
+++ b/src/murmur/Server.cpp
@@ -1047,7 +1047,7 @@ void Server::sendMessage(ServerUser &u, const unsigned char *data, int len, QByt
 			((reinterpret_cast< quint64 >(ebuffer.data()) + 8) & static_cast< quint64 >(~7)) + 4);
 #else
 		std::vector< char > bufVec;
-		bufVec.resize(len + 4);
+		bufVec.resize(static_cast< std::size_t >(len + 4));
 		char *buffer    = bufVec.data();
 #endif
 		{

